src/human-readable/types/signatures.ts

```diff
@@ -128,6 +128,12 @@ export type IsName<TName extends string> = TName extends ''
   : ValidateName<TName> extends TName
   ? true
   : false
+
+export type AssertName<TName extends string> =
+  ValidateName<TName> extends infer InvalidName extends string[]
+    ? InvalidName[number]
```

Comment:
might be nice to contain the error in brackets so it's easier to see the boundary.

```ts
 expectTypeOf<
  FormatAbiParameter<{
    type: 'address'
    name: '123'
  }>
>().toEqualTypeOf<'address [Error: Identifier "123" cannot be a number string.]'>()
```

```suggestion
    ? `[${InvalidName[number]}]`
```